sections/appendix/testplan.tex

```diff
@@ -0,0 +1,60 @@
+\section{Test plan}
+
+\subsection{Documentation}
+All functions should be documented. 
+This is done by writing comments in the code following the standards of the language.
+\\\\
+For C\#: 
+\\
+\url{https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments}
+\\\\
+For Python:
+\\
+\url{https://www.geeksforgeeks.org/python-docstrings/}
+\\\\
+These comments should explain what the parameters of the function are and a short summary of what the function does.
+\\\\
+If the function that is being implemented deals with more complex concepts, additional comments should be written in the code to explain what is happening in the function.
+
+\subsection{Testing}
+\subsubsection{What to test}
+All code that can influence the user's experience should be tested. 
+This means that every function that would influence the program negatively if it had an incorrect calculation should be tested to try to avoid this.
+\\
+All public functions should always be tested.
+
+\subsubsection{How to test}
+
+\begin{itemize}
+    \item Unit tests
+        \begin{itemize}
+            \item Functions are tested in isolation.
+            \item If the function needs data from outside the function it should be manually generated using mocks.
+            \item \href{http://softwaretestingfundamentals.com/unit-testing/}{More information about unit testing}
+        \end{itemize}
+    \item Integration tests
+        \begin{itemize}
+            \item Testing if different modules integrate correctly.
+            \item An example of this is when we are fetching data from the python server into the C\# client.
+            \item \href{http://softwaretestingfundamentals.com/integration-testing/}{More information about integration testing}
+        \end{itemize}
+\end{itemize}
+
+\subsubsection{Unit testing in unity}
+There are two ways of running unit tests in unity which is in play mode and in editor mode. 
+Play mode requires the framework to load the scenes before testing code while editor mode can test the scripts in the project without loading the scenes. 
+When the scenes are loaded in play mode the objects are instantiated with the predefined data in the scene. 
+This makes it a bit harder to control the state of the program while testing.
+\\ 
+The focus will be on testing all of the functions isolated in editor mode. 
```

Comment:
```suggestion
Since this makes it more difficult to control the state of the program, the focus will be on testing the functions isolated in editor mode. 
```